features/dd-sdk-android-webview/src/main/kotlin/com/datadog/android/webview/WebViewTracking.kt

```diff
@@ -76,23 +79,58 @@ object WebViewTracking {
         val featureContext = sdkCore.getFeatureContext(Feature.SESSION_REPLAY_FEATURE_NAME)
         val privacyLevel = (featureContext[SESSION_REPLAY_PRIVACY_KEY] as? String)
             ?: SESSION_REPLAY_MASK_ALL_PRIVACY
-        val webViewEventConsumer = buildWebViewEventConsumer(featureSdkCore, logsSampleRate)
+        val webViewEventConsumer = buildWebViewEventConsumer(webView, featureSdkCore, logsSampleRate)
         webView.addJavascriptInterface(
             DatadogEventBridge(webViewEventConsumer, allowedHosts, privacyLevel),
             DATADOG_EVENT_BRIDGE_NAME
         )
     }
 
     private fun buildWebViewEventConsumer(
+        webView: WebView,
         sdkCore: FeatureSdkCore,
         logsSampleRate: Float
     ): WebViewEventConsumer<String> {
+        val webViewRumFeature = resolveRumFeature(sdkCore)
+        val webViewLogsFeature = resolveLogsFeature(sdkCore)
+        val webViewReplayFeature = resolveReplayFeature(sdkCore)
+        val contextProvider = WebViewRumEventContextProvider(sdkCore.internalLogger)
+        if (webViewReplayFeature == null && webViewRumFeature == null) {
```

Comment:
there is a mistake here, we need also to check `webViewLogsFeature` here. Also it makes sense to create `contentProvider` only inside `else` block.

Reply:
yes something was wrong during a `renaming` :(. Nice spot, I need to see why this was not caught in the unit tests actually.